### PR TITLE
Introduced two new use cases related to API documentation and collection pagination

### PR DESCRIPTION
**Pull request by alien-mcl**
Please find the attached use cases related to:
- API documentation of data structures
- API user documentation
- partial collection handling

---
This change is [Reviewable](https://reviewable.io/reviews/hydracg/specifications/159)

### CODE REVIEW

**Comment by lanthaler** [comment]
Reviewed 4 of 5 files at r1.
Review status: 4 of 5 files reviewed at latest revision, all discussions resolved.

---

*[drafts/use-cases/2.1.api-documentation-data-structures.md, line 1 at r1](https://beta.reviewable.io/reviews/hydracg/specifications/159#-L7TOIWl3Y1qaNywobmq:-L7TOIWl3Y1qaNywobmr:b-mebgkj) ([raw file](https://github.com/hydracg/specifications/blob/9ec1f3c4b27ee71e17c55325465f43f85e77ffe3/drafts/use-cases/2.1.api-documentation-data-structures.md#L1)):*
> documentationt

Typo --> documentation

---

*[drafts/use-cases/2.1.api-documentation-data-structures.md, line 27 at r1](https://beta.reviewable.io/reviews/hydracg/specifications/159#-L7TOdUC2XxJbkr-kai6:-L7TOUHpD-s8LLGDs87p:bmzf8pb) ([raw file](https://github.com/hydracg/specifications/blob/9ec1f3c4b27ee71e17c55325465f43f85e77ffe3/drafts/use-cases/2.1.api-documentation-data-structures.md#L27)):*
>  There are situations, when i.e. Rich...

should be "There are situations when, e.g., Rich..."

---

*[drafts/use-cases/2.1.api-documentation-data-structures.md, line 33 at r1](https://beta.reviewable.io/reviews/hydracg/specifications/159#-L7TOsoo-Tew7_biEHOj:-L7TOsoo-Tew7_biEHOk:b-evssbv) ([raw file](https://github.com/hydracg/specifications/blob/9ec1f3c4b27ee71e17c55325465f43f85e77ffe3/drafts/use-cases/2.1.api-documentation-data-structures.md#L33)):*
> could be

--> is

---

*[drafts/use-cases/2.1.api-documentation-data-structures.md, line 34 at r1](https://beta.reviewable.io/reviews/hydracg/specifications/159#-L7TOuG99UG31b3lH4uu:-L7TOuG99UG31b3lH4uv:b46yxaj) ([raw file](https://github.com/hydracg/specifications/blob/9ec1f3c4b27ee71e17c55325465f43f85e77ffe3/drafts/use-cases/2.1.api-documentation-data-structures.md#L34)):*
> vs.

--> and

---

*[drafts/use-cases/2.1.api-documentation-data-structures.md, line 44 at r1](https://beta.reviewable.io/reviews/hydracg/specifications/159#-L7TP-83BXSVMKMmo54r:-L7TP-83BXSVMKMmo54s:b-zhf62m) ([raw file](https://github.com/hydracg/specifications/blob/9ec1f3c4b27ee71e17c55325465f43f85e77ffe3/drafts/use-cases/2.1.api-documentation-data-structures.md#L44)):*
> api?documentation

Nit: While it wouldn't matter in practice, I think people will find the query parameter confusing. I'd thus suggest to place this at `/api/documentation`

---

*[drafts/use-cases/2.1.api-documentation-data-structures.md, line 58 at r1](https://beta.reviewable.io/reviews/hydracg/specifications/159#-L7TPPiy91RhGFo6JUbh:-L7TPPiy91RhGFo6JUbi:b6d9312) ([raw file](https://github.com/hydracg/specifications/blob/9ec1f3c4b27ee71e17c55325465f43f85e77ffe3/drafts/use-cases/2.1.api-documentation-data-structures.md#L58)):*
> ```Markdown
> {
>     "@context": "/api/context.jsonld",
>     "@id": "/api?documentation",
> ```

Please type it as `hydra:ApiDocumentation`

---

*[drafts/use-cases/2.1.api-documentation-data-structures.md, line 64 at r1](https://beta.reviewable.io/reviews/hydracg/specifications/159#-L7TPUXi9P7tLTCBuxAU:-L7TPUXi9P7tLTCBuxAV:bm2d8i9) ([raw file](https://github.com/hydracg/specifications/blob/9ec1f3c4b27ee71e17c55325465f43f85e77ffe3/drafts/use-cases/2.1.api-documentation-data-structures.md#L64)):*
> ```Markdown
>             "supportedProperty": [
>                 {
>                     "property": "schema:eventName",
> ```

These all need to be separate objects (of type `hydra:SupportedProperty`)

---

*[drafts/use-cases/2.2.api-documentation-user-document.md, line 23 at r1](https://beta.reviewable.io/reviews/hydracg/specifications/159#-L7TPzMX3zglhbHuJl3g:-L7TPzMX3zglhbHuJl3h:b-wz825f) ([raw file](https://github.com/hydracg/specifications/blob/9ec1f3c4b27ee71e17c55325465f43f85e77ffe3/drafts/use-cases/2.2.api-documentation-user-document.md#L23)):*
> foreach (const operation of supportedProperty.supportedOperations)

Did you want to add some inner statements?

---

*[drafts/use-cases/3.2.pagination.md, line 35 at r1](https://beta.reviewable.io/reviews/hydracg/specifications/159#-L7TQHzC8rLcL0Y-LZWl:-L7TQHzC8rLcL0Y-LZWm:bthhxmv) ([raw file](https://github.com/hydracg/specifications/blob/9ec1f3c4b27ee71e17c55325465f43f85e77ffe3/drafts/use-cases/3.2.pagination.md#L35)):*
> Cleint

--> Client

---


*Comments from [Reviewable](https://beta.reviewable.io/reviews/hydracg/specifications/159)*

**Comment by alien-mcl** [comment]
Review status: 4 of 5 files reviewed at latest revision, 9 unresolved discussions.

---

*[drafts/use-cases/2.1.api-documentation-data-structures.md, line 1 at r1](https://reviewable.io:443/reviews/hydracg/specifications/159#-L7TOIWl3Y1qaNywobmq:-L7VtiAD702BcMwkglZU:b-896fix) ([raw file](https://github.com/hydracg/specifications/blob/9ec1f3c4b27ee71e17c55325465f43f85e77ffe3/drafts/use-cases/2.1.api-documentation-data-structures.md#L1)):*
<details><summary><i>Previously, lanthaler (Markus Lanthaler) wrote…</i></summary><blockquote>

> documentationt

Typo --> documentation
</blockquote></details>

Done.

---

*[drafts/use-cases/2.1.api-documentation-data-structures.md, line 27 at r1](https://reviewable.io:443/reviews/hydracg/specifications/159#-L7TOdUC2XxJbkr-kai6:-L7VtpI75_CUmcUo7d4Y:b-896fix) ([raw file](https://github.com/hydracg/specifications/blob/9ec1f3c4b27ee71e17c55325465f43f85e77ffe3/drafts/use-cases/2.1.api-documentation-data-structures.md#L27)):*
<details><summary><i>Previously, lanthaler (Markus Lanthaler) wrote…</i></summary><blockquote>

>  There are situations, when i.e. Rich...

should be "There are situations when, e.g., Rich..."
</blockquote></details>

Done.

---

*[drafts/use-cases/2.1.api-documentation-data-structures.md, line 33 at r1](https://reviewable.io:443/reviews/hydracg/specifications/159#-L7TOsoo-Tew7_biEHOj:-L7VttFUa2aCgo3_lgpR:b-896fix) ([raw file](https://github.com/hydracg/specifications/blob/9ec1f3c4b27ee71e17c55325465f43f85e77ffe3/drafts/use-cases/2.1.api-documentation-data-structures.md#L33)):*
<details><summary><i>Previously, lanthaler (Markus Lanthaler) wrote…</i></summary><blockquote>

> could be

--> is
</blockquote></details>

Done.

---

*[drafts/use-cases/2.1.api-documentation-data-structures.md, line 34 at r1](https://reviewable.io:443/reviews/hydracg/specifications/159#-L7TOuG99UG31b3lH4uu:-L7VtwR1QPG05qCZs0dA:b-896fix) ([raw file](https://github.com/hydracg/specifications/blob/9ec1f3c4b27ee71e17c55325465f43f85e77ffe3/drafts/use-cases/2.1.api-documentation-data-structures.md#L34)):*
<details><summary><i>Previously, lanthaler (Markus Lanthaler) wrote…</i></summary><blockquote>

> vs.

--> and
</blockquote></details>

Done.

---

*[drafts/use-cases/2.1.api-documentation-data-structures.md, line 44 at r1](https://reviewable.io:443/reviews/hydracg/specifications/159#-L7TP-83BXSVMKMmo54r:-L7Vu5MB8GoDReVv3ggb:bd9yjm7) ([raw file](https://github.com/hydracg/specifications/blob/9ec1f3c4b27ee71e17c55325465f43f85e77ffe3/drafts/use-cases/2.1.api-documentation-data-structures.md#L44)):*
<details><summary><i>Previously, lanthaler (Markus Lanthaler) wrote…</i></summary><blockquote>

> api?documentation

Nit: While it wouldn't matter in practice, I think people will find the query parameter confusing. I'd thus suggest to place this at `/api/documentation`
</blockquote></details>

Hmm. Actually I didn't want the documentation to be confused with an actual API resource as other (`/api/event` to not to look to far) are addressed same way. Do we have another option?

---

*[drafts/use-cases/2.1.api-documentation-data-structures.md, line 58 at r1](https://reviewable.io:443/reviews/hydracg/specifications/159#-L7TPPiy91RhGFo6JUbh:-L7VujXH7ij5CSYRuUqx:b-896fix) ([raw file](https://github.com/hydracg/specifications/blob/9ec1f3c4b27ee71e17c55325465f43f85e77ffe3/drafts/use-cases/2.1.api-documentation-data-structures.md#L58)):*
<details><summary><i>Previously, lanthaler (Markus Lanthaler) wrote…</i></summary><blockquote>

Please type it as `hydra:ApiDocumentation`
</blockquote></details>

Done.

---

*[drafts/use-cases/2.1.api-documentation-data-structures.md, line 64 at r1](https://reviewable.io:443/reviews/hydracg/specifications/159#-L7TPUXi9P7tLTCBuxAU:-L7Vv0wWDDtIvYMVTFl9:b4kfzk7) ([raw file](https://github.com/hydracg/specifications/blob/9ec1f3c4b27ee71e17c55325465f43f85e77ffe3/drafts/use-cases/2.1.api-documentation-data-structures.md#L64)):*
<details><summary><i>Previously, lanthaler (Markus Lanthaler) wrote…</i></summary><blockquote>

These all need to be separate objects (of type `hydra:SupportedProperty`)
</blockquote></details>

Argh - correct. I wanted it to be shorter and it ended up with some virtual stupidity. Done

---

*[drafts/use-cases/2.2.api-documentation-user-document.md, line 23 at r1](https://reviewable.io:443/reviews/hydracg/specifications/159#-L7TPzMX3zglhbHuJl3g:-L7Vvw7VyAKNNB6K0Vb-:b37ramf) ([raw file](https://github.com/hydracg/specifications/blob/9ec1f3c4b27ee71e17c55325465f43f85e77ffe3/drafts/use-cases/2.2.api-documentation-user-document.md#L23)):*
<details><summary><i>Previously, lanthaler (Markus Lanthaler) wrote…</i></summary><blockquote>

> foreach (const operation of supportedProperty.supportedOperations)

Did you want to add some inner statements?
</blockquote></details>

Ahh, yep - jumped to another part of document and forgot to get back here. Done

---

*[drafts/use-cases/3.2.pagination.md, line 35 at r1](https://reviewable.io:443/reviews/hydracg/specifications/159#-L7TQHzC8rLcL0Y-LZWl:-L7Vw3tjXYooSNfiOU9z:b-896fix) ([raw file](https://github.com/hydracg/specifications/blob/9ec1f3c4b27ee71e17c55325465f43f85e77ffe3/drafts/use-cases/3.2.pagination.md#L35)):*
<details><summary><i>Previously, lanthaler (Markus Lanthaler) wrote…</i></summary><blockquote>

> Cleint

--> Client
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/hydracg/specifications/159)*

**Comment by lanthaler** [comment]
Review status: 4 of 5 files reviewed at latest revision, 9 unresolved discussions.

---

*[drafts/use-cases/2.1.api-documentation-data-structures.md, line 1 at r1](https://reviewable.io:443/reviews/hydracg/specifications/159#-L7TOIWl3Y1qaNywobmq:-L7YK3ETqc5cWy9GT-9y:b6ntv03) ([raw file](https://github.com/hydracg/specifications/blob/9ec1f3c4b27ee71e17c55325465f43f85e77ffe3/drafts/use-cases/2.1.api-documentation-data-structures.md#L1)):*
<details><summary><i>Previously, alien-mcl (Karol Szczepański) wrote…</i></summary><blockquote>

Done.
</blockquote></details>

Looks like you forgot to push your changes.

---

*[drafts/use-cases/2.1.api-documentation-data-structures.md, line 44 at r1](https://reviewable.io:443/reviews/hydracg/specifications/159#-L7TP-83BXSVMKMmo54r:-L7YK97YzBiiWHJcw1Lo:bw5no1p) ([raw file](https://github.com/hydracg/specifications/blob/9ec1f3c4b27ee71e17c55325465f43f85e77ffe3/drafts/use-cases/2.1.api-documentation-data-structures.md#L44)):*
<details><summary><i>Previously, alien-mcl (Karol Szczepański) wrote…</i></summary><blockquote>

Hmm. Actually I didn't want the documentation to be confused with an actual API resource as other (`/api/event` to not to look to far) are addressed same way. Do we have another option?
</blockquote></details>

What about `/api-documentation` then? It is an actual API resource btw. ;-) A crucial part of its interface

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/hydracg/specifications/159)*

**Comment by alien-mcl** [comment]
Review status: 2 of 5 files reviewed at latest revision, 9 unresolved discussions.

---

*[drafts/use-cases/2.1.api-documentation-data-structures.md, line 1 at r1](https://reviewable.io:443/reviews/hydracg/specifications/159#-L7TOIWl3Y1qaNywobmq:-L7a70hdnf5xIYEdXcp-:bh46iwj) ([raw file](https://github.com/hydracg/specifications/blob/9ec1f3c4b27ee71e17c55325465f43f85e77ffe3/drafts/use-cases/2.1.api-documentation-data-structures.md#L1)):*
<details><summary><i>Previously, lanthaler (Markus Lanthaler) wrote…</i></summary><blockquote>

Looks like you forgot to push your changes.
</blockquote></details>

Nope - looks like Reviable misbehaves. See (this)[https://github.com/HydraCG/Specifications/pull/159/files#diff-5d0fa46b254a904910d47b58d99d20d8] file - it's correct now.

---

*[drafts/use-cases/2.1.api-documentation-data-structures.md, line 44 at r1](https://reviewable.io:443/reviews/hydracg/specifications/159#-L7TP-83BXSVMKMmo54r:-L7a7BzLnV4XddV52jk4:b3w80ll) ([raw file](https://github.com/hydracg/specifications/blob/9ec1f3c4b27ee71e17c55325465f43f85e77ffe3/drafts/use-cases/2.1.api-documentation-data-structures.md#L44)):*
<details><summary><i>Previously, lanthaler (Markus Lanthaler) wrote…</i></summary><blockquote>

What about `/api-documentation` then? It is an actual API resource btw. ;-) A crucial part of its interface
</blockquote></details>

Well, `/api-documentation` feels better.

I understand that that documentation is a crutial part of its interface, but I wouldn't like to see the documentation mixed with other business-logic related resources.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/hydracg/specifications/159)*

**Comment by lanthaler** [comment]
:lgtm: thanks!

---

Reviewed 2 of 3 files at r2.
Review status: 4 of 5 files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/hydracg/specifications/159#-:-L7chUwl7coMbiypk48j:brfqlux)*

**Comment by lanthaler** [comment]
Reviewed 1 of 3 files at r2.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/hydracg/specifications/159)*

File: drafts/use-cases/3.2.pagination.md [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+    "@context": "/api/context.jsonld",
+    "@id": "/api/events",
+    "@type": "PartialCollection",
+    "manages": {
+      "property": "rdf:type",
+      "object": "schema:Event"
+    },
+    "next": "/api/events?page=2",
+    "first": "/api/events",
+    "last": "/api/events?page=2",
```
[the remark this answers is not in the document]
Comment (tpluscode) [reply drafts/use-cases/3.2.pagination.md]: I think you're right. The `PartialCollectionView` is the official design. A new issue sounds like the way to go

**Pull request merged**